include/range/v3/view/single.hpp

```diff
@@ -38,20 +39,20 @@ namespace ranges
         {
         private:
             friend struct range_access;
-            Val value_;
+            optional<Val> value_opt_;
             struct cursor
             {
             private:
-                Val value_;
+                optional<Val> value_opt_;
```

Comment:
This should use the `semiregular_t` helper instead of using `optional` directly. See how the other views do it; for example, `istream_view`.